autoload/clap/cache.vim

```diff
@@ -0,0 +1,44 @@
+" Author: Mark Wu <markplace@gmail.com>
+" Description: Cache API for clap.
+
+let s:save_cpo = &cpoptions
+set cpoptions&vim
+
+let s:path_separator = has('win32') ? '\' : '/'
+let s:clap_cache_directory = get(g:, 'clap_cache_directory', '')
+
+function! clap#cache#directory() abort
+  if empty(s:clap_cache_directory)
+    if has('nvim')
+      let user_cache = stdpath('cache')
+    elseif exists('$XDG_CACHE_HOME')
+      let user_cache = $XDG_CACHE_HOME
+    else
+      let user_cache = $HOME . s:path_separator . '.cache'
+    endif
+    let s:clap_cache_directory = user_cache . s:path_separator . 'clap'
+  endif
+
+  if !isdirectory(s:clap_cache_directory)
+    call mkdir(s:clap_cache_directory, 'p')
+  endif
+
+  return s:clap_cache_directory
+endf
+
+function! clap#cache#location_for(provider_id, fname) abort
+  if empty(a:provider_id)
+    let provider_cache_directory = clap#cache#directory()
```

Comment:
I suggest not allowing passing a empty `provider_id` in which case it makes no differences to `clap#cache#directory()` and it's not recommended.

Reply:
done!